plugins/kubernetes-backend/src/kubernetes-auth-translator/AzureIdentityKubernetesAuthTranslator.ts

```diff
@@ -31,11 +41,23 @@ export class AzureIdentityKubernetesAuthTranslator
       clusterDetails,
     );
 
-    const credentials = new DefaultAzureCredential();
+    if (this.tokenExpired()) {
+      this.accessToken = await this.tokenCredential.getToken(aksScope);
+
+      if (!this.accessToken) {
+        throw new Error('Unable to retrieve Azure token');
+      }
+    }
 
-    // TODO: can we cache this? It's inneficiant to get a new token every time
-    const accessToken = await credentials.getToken(aksScope);
-    clusterDetailsWithAuthToken.serviceAccountToken = accessToken.token;
+    clusterDetailsWithAuthToken.serviceAccountToken = this.accessToken!.token;
```

Comment:
Yep `!` is always a bit ugly :grin: Probably better keep the check as it was then. Another option could be to extract the access token logic into a `this.getAccessToken()` method, which takes care of both checking for expiry and issuing the new token.